### ISSUE TITLE
Accessibility testing is needed for {learnr} widgets

### ISSUE THREAD

**Issue by jooyoungseo** [rstudio/learnr#636]
As this title suggests, {learnr} seems to lack web accessibility components. As {learnr} tutorials are increasingly being used in formal and informal educational settings, accessibility compliance is critical to ensure everyone's equal access.

Do you have any plans to improve accessibility for {learnr} widgets?

**Comment by jooyoungseo** [comment]
Awesome! I see the improvement there for landmark/skip navigation! Thanks for your great work, @gadenbuie!

I still don't see any interactive control when using screen reader in the following example. I am not sure whether this is a good example though for a11y test.

````
---
title: "Hello, Tutorial!"
output: learnr::tutorial
runtime: shiny_prerendered
---

```{r setup, include=FALSE}
library(learnr)
```

This code computes the answer to one plus one,
change it so it computes two plus two:

```{r addition, exercise=TRUE}
1 + 1
```
````

**Comment by jooyoungseo** [comment]
@gadenbuie -- Currently, landmark/skip navigation, and heading navigation are not the critical issues because we can simply fix/add semantics using aria roles. However, as you pointed out, the editor is the pain. I don't see any editable text box in the example above when using screen reader. As far as an accessible editor is concerned, I would suggest using [Monaco editor](https://microsoft.github.io/monaco-editor/) used in VSCode, which is the most accessible code editor that I have tested. It has a a11y mode which you can find more info [here](https://github.com/microsoft/monaco-editor/wiki/Monaco-Editor-Accessibility-Guide).

**Comment by jooyoungseo** [comment]
CodeMirror used to be not that accessible, but it seems like it has improved a lot over the recent years. I have just tested it now, and it works much better than before. Yes, I think CodeMirror could be another good choice.
